frontend/packages/knative-plugin/src/components/add/event-sources/EventSourcesSelector.tsx

```diff
@@ -1,24 +1,42 @@
 import * as React from 'react';
-import { useFormikContext, FormikValues } from 'formik';
+import { useFormikContext, FormikValues, useField } from 'formik';
+import { safeDump } from 'js-yaml';
 import { ItemSelectorField } from '@console/shared';
 import FormSection from '@console/dev-console/src/components/import/section/FormSection';
-import { NormalizedEventSources } from '../import-types';
-import { getEventSourceData } from '../../../utils/create-eventsources-utils';
+import { NormalizedEventSources, EditorType } from '../import-types';
+import {
+  getEventSourceData,
+  getEventSourceYaml,
+  isKnownEventSourceSelected,
+} from '../../../utils/create-eventsources-utils';
 
 interface EventSourcesSelectorProps {
   eventSourceList: NormalizedEventSources;
 }
 
 const EventSourcesSelector: React.FC<EventSourcesSelectorProps> = ({ eventSourceList }) => {
   const { setFieldValue, setFieldTouched, validateForm } = useFormikContext<FormikValues>();
+  const [namespaceField] = useField('project.name');
   const handleItemChange = React.useCallback(
     (item: string) => {
-      const name = `data.${item.toLowerCase()}`;
-      setFieldValue(name, getEventSourceData(item.toLowerCase()));
-      setFieldTouched(name, true);
+      if (isKnownEventSourceSelected(item)) {
+        const name = `data.${item.toLowerCase()}`;
+        setFieldValue(name, getEventSourceData(item.toLowerCase()));
+        setFieldTouched(name, true);
+        setFieldValue('editorType', EditorType.Form);
+        setFieldTouched('editorType', true);
+      } else {
+        setFieldValue(
+          'yamlData',
+          safeDump(getEventSourceYaml(namespaceField.value, item.toLowerCase())),
+        );
```

Comment:
same as above , I think we shouldn't have it here , maybe it's good to introduce a prop on this component `onChange` and handle these logic in callback in `eventSourceForm` and access from values

Reply:
We can simply move the whole loading yaml logic in `YAMLSection` itself. Just wrap this logic inside a `useEffect` with dependency on `values.type`. It will load the YAML if the section is rendered and when the type changes automatically.

Reply:
moved the logic to `eventSourceForm` and passed down to `EventSourceSelector` as a prop.
it handles both the form and yaml editor cases.
should we move just the yaml logic and keep the form logic in `handleChange` ?